orttraining/orttraining/python/training/_ortmodule_execution_manager.py

```diff
@@ -0,0 +1,268 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+from . import _utils, _ortmodule_utils, _ortmodule_output_transformation as _ortmodule_io
+
+from onnxruntime.capi.onnxruntime_inference_collection import OrtValue
+from onnxruntime.capi import _pybind_state as C
+
+import io
+import inspect
+import onnx
+import onnxruntime
+import torch
+from torch.utils.cpp_extension import ROCM_HOME
+
+ONNX_OPSET_VERSION = 12
+
+class ExecutionManager(object):
```

Comment:
```python
'''What is a Execution manager and what is the problem it solves?
is it used directly or should be extended, etc
which classes does it interact with (ExecutionAgent)?
'''
```
docstring for this class is important due to its central role in the problem. Contributors need to understand would the big blocks play together

Reply:
Maybe call it `GraphExecutionManager` to clarify what it will execute. "Execution" might be confused to a python script, an OS application, etc